### PR TITLE
Integrate the Exometer metrics package

### PR DESCRIPTION
**Pull request by uwiger**
[Exometer](https://github.com/Feuerlabs/exometer) is a metrics application developed for Basho by Feuerlabs. Among other things, it offers:
- More dynamic management of metrics (including enabling/disabling metrics at runtime)
- More cost-effective histograms
- Custom metrics
- Reporters to external services such as Graphite and Collectd

This pull request depends on a number of other PRs, i.e. basho/riak_core#465, basho/riak_kv#748, basho/riak_api#45, basho/riak_pipe#81, basho/riak_control#175, basho/riak_search#155, basho/yokozuna#246, basho/sidejob#4

According to our benchmarks, this PR slightly improves throughput and latency:

summary-exo

compared to the starting point (the develop branch as of Dec 3):

summary-orig

Since it is possible to selectively and dynamically disable metrics, throughput can be further increased at the expense of metrics information.

Memory footprint has also been reduced, likely due to less caching and more economical histograms:

memory-usage-exo

compared to the starting point:

memory-usage-orig

### CODE REVIEW

File: rel/reltool.config [hunk 1]
```diff
@@ -107,9 +110,11 @@
            {template, "../deps/riak_sysmon/priv/riak_sysmon.schema", "lib/15-riak_sysmon.schema"},
            {template, "../deps/bitcask/priv/bitcask.schema", "lib/16-bitcask.schema"},
            {template, "../deps/riak_control/priv/riak_control.schema", "lib/17-riak_control.schema"},
+           {template, "../deps/exometer/priv/exometer.schema", "lib/18-exometer.schema"},
 
            {template, "../deps/riak_kv/priv/multi_backend.schema", "lib/20-multi_backend.schema"},
            {template, "../deps/eleveldb/priv/eleveldb.schema", "lib/21-leveldb.schema"},
+           {template, "../deps/sidejob/priv/sidejob.schema", "lib/22-sidejob.schema"},
```
Comment (evanmcc) [inline rel/reltool.config]: this line needs to be removed if we're removing exometer/sidejob stuff.

Comment (uwiger) [inline rel/reltool.config]: Apologies. I had done that, but not pushed it. Pushed now.

2013/12/6 Evan Vigil-McClanahan notifications@github.com

**Comment by subnetmarco** [comment]
+1

**Comment by jburwell** [comment]
Superseded by https://github.com/basho/riak/pull/621

**Pull request closed without merging**